Wavenet.Api.ViewModels.IdViewModel/IdViewModel.cs

```diff
@@ -149,7 +149,7 @@ public IdViewModel(string code)
         public bool Equals(IdViewModel other) => this.Id == other?.Id;
 
         /// <inheritdoc />
-        public override bool Equals(object obj) => this.Equals(obj as IdViewModel);
+        public override bool Equals(object obj) => !(obj is null) && this.Equals(obj as IdViewModel);
```

Comment:
Je pense reprendre l'idée mais l'adapter en:
```suggestion
        public override bool Equals(object obj) => obj is IdViewModel id && this.Equals(id);
```

Ce qui devrait éviter le call a Equals en cas de null mais aussi dans le cas ou obj n'est pas du bon type.